packages/nextjs/src/edge/wrapAppDirComponentWithSentry.ts

```diff
@@ -0,0 +1,32 @@
+import { captureException } from '@sentry/core';
+
+/**
+ * Wraps an `app` directory server component with Sentry error instrumentation.
+ */
+export function wrapAppDirComponentWithSentry<F extends (...args: any[]) => any>(appDirComponent: F): F {
+  // Even though users may define server components as async functions, for the client bundles
+  // Next.js will turn them into synchronous functionsf and it will transform any`await`s into instances of the`use`
```

Comment:
I wonder what kind of spans we could get from here 🤔

Reply:
We could probably get some nice ones but since async context is super funky here atm with the app dir we can't have only one transaction for all the server components and I would feel bad for creating individual transactions for each server component.

(We need to put in some more work here - I will create an issue to track everything that we still can improve)